preparation/demo/src/service/BlockSchema.js

```diff
@@ -0,0 +1,43 @@
+import mongoose from 'mongoose';
+
+const BlockSchema = new mongoose.Schema({
+  prev_hash : {
+    type : String,
+    required: true
+  },
+  total_gas : {
+    type : Number,
+    required : true
+  },
+  total_credit : {
+    type : Number,
+    required : true
+  },
+  tx_log : {
+    type : [String],
+    required : true
+  },
+  block_owner : {
+    type : String,
+    required : true
+  },
+  height : {
+    type : Number,
+    required : true
+  },
+  hash : {
+    type : String,
+    required : true
+  },
+  
+
+
+}, {
+  timestamps: true,
+});
+
+BlockSchema.statics = {
+  
+};
+
```

Comment:
We need a state machine in the block to store a Map<ownerPeerId -> creditScore> and Map<ownerPeerId -> gasBalance>
Also a n array of tx in this block etc.

This block can be saved to IPFS because its size. The IPFS Hash (called CID) can be broadcast in the P2P network so that every client can reeive the CID, then retreive from IPFS. 

In the future version, we should use IPFS DAG to store the block. See this demo:  https://github.com/ipfs/js-ipfs/tree/master/examples/explore-ethereum-blockchain